models/DataObject/Service.php

```diff
@@ -139,6 +139,10 @@ public function copyRecursive($target, $source)
         //load all in case of lazy loading fields
         self::loadAllObjectFields($source);
 
+        // triggers actions before object cloning
+        $event = new DataObjectEvent($source);
```

Comment:
Should we also trigger the event when copying children?

Reply:
@dturopoli 🏓

Reply:
Post copy event is already also triggering on copying children so I just used the same logic for pre copy

Reply:
you are right, then it should be consistent with post copy

Reply:
One more thing, like post copy contains `$new` as main element and passes `$source` as `base_element` in params. I think then it makes sense to pass `$target` as `target_element` in params of pre copy. Then you would know what will be the target and work with that info. wdyt?

Reply:
I agree

Reply:
could you please change it? 😄

Reply:
Added target_element